presto-spi/src/main/java/com/facebook/presto/spi/Plugin.java

```diff
@@ -13,12 +13,25 @@
  */
 package com.facebook.presto.spi;
 
+import com.facebook.presto.spi.block.BlockEncodingSerde;
+import com.facebook.presto.spi.type.TypeManager;
+
 import java.util.List;
 import java.util.Map;
 
 public interface Plugin
 {
     default void setOptionalConfig(Map<String, String> optionalConfig) {}
 
+    default void setTypeManager(TypeManager typeManager) {}
+
+    default void setNodeManager(NodeManager nodeManager) {}
+
+    default void setBlockEncodingSerde(BlockEncodingSerde blockEncodingSerde) {}
```

Comment:
I agree, some of these should become connector specific, but that's a bigger change. This is the first step to get rid of injection (and make sure nothing breaks). I'll add a context object to ConnectorFactory later and move some of these.